external/owee/owee_archive.ml

```diff
@@ -0,0 +1,288 @@
+          (* Strip trailing nulls from the name *)
+          let name =
+            match String.index_opt name '\000' with
+            | Some i -> String.sub name 0 i
+            | None -> name
+          in
+          name, after_header + name_len, total_size - name_len
+        | None -> (
+          match parse_sysv_extended_name ar_name with
+          | Some offset ->
+            (* System V extended filename: look up in string table *)
+            let name =
+              match string_table with
+              | None ->
+                invalid_format
+                  (Printf.sprintf
+                     "Extended filename at offset %d but no string table found"
+                     header_start)
+              | Some strtab -> read_sysv_string_table_entry strtab offset
+            in
+            name, after_header, total_size
+          | None ->
+            (* Regular short filename *)
+            let trimmed = String.trim ar_name in
+            let name =
+              (* Strip trailing '/' if present (System V style) *)
+              if String.ends_with ~suffix:"/" trimmed
+              then String.sub trimmed 0 (String.length trimmed - 1)
+              else trimmed
+            in
+            name, after_header, total_size)
+      in
+      (* Skip past the member data for the next iteration *)
+      advance cursor total_size;
+      Some
+        { name;
+          size;
+          data_offset;
+          mtime = parse_decimal_field ar_date;
+          uid = parse_decimal_field ar_uid;
+          gid = parse_decimal_field ar_gid;
+          mode = parse_octal_field ar_mode
+        }))
+
+(* Check if a member is a special entry that should be skipped *)
+let is_special_member member =
+  is_sysv_symtab member.name || is_sysv_strtab member.name
+  || is_bsd_symtab member.name
+
+let read buf =
+  let cur = Owee_buf.cursor buf in
+  read_magic cur;
+  (* First pass: look for the System V string table "//" if present. BSD
+     archives don't have a separate string table. *)
+  let string_table = ref None in
+  let first_pass_cursor = Owee_buf.cursor buf ~at:archive_magic_size in
+  let rec find_string_table () =
+    if at_end first_pass_cursor
+    then ()
+    else (
+      if first_pass_cursor.position mod 2 = 1 then advance first_pass_cursor 1;
+      if at_end first_pass_cursor
+      then ()
+      else (
+        ensure first_pass_cursor header_size "Truncated header in first pass";
+        let ar_name = Read.fixed_string first_pass_cursor 16 in
+        let _ = Read.fixed_string first_pass_cursor 12 in
+        (* date *)
+        let _ = Read.fixed_string first_pass_cursor 6 in
+        (* uid *)
+        let _ = Read.fixed_string first_pass_cursor 6 in
+        (* gid *)
+        let _ = Read.fixed_string first_pass_cursor 8 in
+        (* mode *)
+        let ar_size = Read.fixed_string first_pass_cursor 10 in
+        let _ = Read.fixed_string first_pass_cursor 2 in
+        (* fmag *)
```

Comment:
The comments appear to refer to the
previous line; I think it would be better
to rather write `let _xyz = ... in`.